submissions/DaniaB24/friends-app/js/app.js

```diff
@@ -0,0 +1,94 @@
+const cardsOfFriends = document.getElementById("allHumans");
+const searchBar = document.getElementById("search");
+const filterMenu = document.getElementById("filter_menu");
+const URL = "https://randomuser.me/api/?results=10";
+let friends;
+
+function startApp() {
+  fetch(URL)
+    .then((response) => response.json())
+    .then((users) => {
+      console.log("ok");
+      friends = users.results;
+      displayUsers(friends);
+      filterMenu.addEventListener("click", (event) => {
+        renderFriends(event, friends);
+      });
+    })
+    .catch((error) => console.log("Authorization failed : " + error.message));
+}
+
+function renderFriends({ target }, friends) {
+  const sortedResults = sortFriends(target, friends);
+  filterByGender(sortedResults ?? []);
+}
+function displayUsers(friends) {
+  const createCards = friends
+    .map((item) => {
+      return `
+      <div class="card_person">
+      <div class="card_container">
+      <img src="${item.picture.large}">
+      <h3 class="card_h3">${item.name.first} ${item.name.last}</h3>
+      <p class="card_age">Age: ${item.dob.age}</p>
+      <p class="card_gender">Gender: ${item.gender}</p>
+      </div>
+      </div>`;
+    })
+    .join("");
+  cardsOfFriends.innerHTML = createCards;
+}
+
+function sortByAge(a, b) {
+  return a.dob.age > b.dob.age ? 1 : -1;
+}
+function sortByName (a, b) {
+  return a.name.first.localeCompare(b.name.first);
+}
+function sortFriends(target, friends) {
+  const button = target.id;
+
+  switch (button) {
+    case "a-z":
+      friends.sort((a, b) => sortByName(a, b));
+      break;
+    case "z-a":
+      friends.sort((a, b) => sortByName(b, a));
+      break;
+    case "0-9":
+      friends.sort((a, b) => sortByAge(a, b));
+
+      break;
+
+    case "9-0":
+      friends.sort((a, b) => sortByAge(b, a));
+
+      break;
+  }
+  return friends;
+}
+function sortByAge(a, b) {
+  return a.dob.age > b.dob.age ? 1 : -1;
+}
+function filterByGender(friends) {
+  const checkboxes = Array.from(
+    document.querySelectorAll(".gender_checkbox:checked")
+  );
+  const filteredUsers = friends.filter((item) =>
+    checkboxes.some((gender) => item.gender === gender.value)
+  );
+  displayUsers(filteredUsers);
+  searchFriends(filteredUsers);
+}
+
+function searchFriends(filteredUsers) {
```

Comment:
It's not running the search in friends. It adds listener to dom element. And render items.

1. Single responsibility
2. Naming

Reply:
And also, every time when you will call this function you will add a new listener. Ouch. Because you will run multiple renders on "every" input event.